### PR TITLE
morse decode

### PR DESCRIPTION
**Pull request by Surafels**
Hello 👋

**Implemented tasks:**

- Create a method to decode a Morse code character, take a string parameter, and return the corresponding character in uppercase .
- Create a method to decode the entire message in Morse code, take a string parameter, and return the string representation.
- Create a method to decode an entire word in Morse code, take a string parameter, and return the string representation.

### CODE REVIEW

**Review by V-Blaze — CHANGES_REQUESTED** [review]
Hi @Surafels 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

dcentl8-9b96e129-9725-401b-ac34-2feb990202a0

## Invalid Code Review Request

## You have submitted a project, that doesn't have the commit of your partner. This is a pair programming activity and therefore, Both students need to contribute meaningfully to this branch. NB: updating readme only is not a valid contribution!

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: morse_code.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+def decode_message(morse_message)
+  morse_message.split.map { |morse_word| decode_word(morse_word) }.join
+end
+
+puts decode_char('.-')
+puts decode_word('-- -.--')
+
+message = '-- -.--   -. .- -- .'
+decoded_message = decode_message(message)
+puts decoded_message
```
Comment (TadesseDev) [inline morse_code.rb]: - [ ] Dear Team, You have done great with the functionalities, I noticed that the last output does not contain space-separated words. when there is more than one word, we need to ensure a space separates the words.  Please consider updating this. 

### The issue 
Screenshot 2023-11-16 at 12 08 50 in the afternoon

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] The decode functionalities are working well ✅
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse_code.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  CODE_CHAR[morse_code] || ''
+end
+
+def decode_word(morse_word)
+  morse_word.split.map { |morse_char| decode_char(morse_char) }.join
+end
+
+def decode_message(morse_message)
+  morse_message.split.map { |morse_word| decode_word(morse_word) }.join
+end
```
Comment (ghost) [inline morse_code.rb]: - [ ] Hey team! 👋 I spotted that there's still an issue with the word spacing. To get the right outcome, consider tweaking your `decode_message` method to look something like this:

```
def decode_message(morse_message)
  morse_words = morse_message.split('   ') # Split by three spaces to separate words
  decoded_words = morse_words.map { |morse_word| decode_word(morse_word) }
  decoded_words.join(' ') # Join the decoded words with a single space
end

```

#### Results

```
"A"
"MY"
"MY NAME"
"A BOX FULL OF RUBIES"

```

#### This should help achieve the desired spacing between words. Keep up the good work! 😊✨

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ `#decode_char` and `#decode_word` works as expected 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @Surafels and @yayaismaail , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: morse decoded


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**